packages/agent/src/agent/http/index.ts

```
@@ -949,7 +961,13 @@ export class HttpAgent implements Agent {
     // Attempt to make the query i=retryTimes times
     // Make query and fetch subnet keys in parallel
     try {
-      const [queryResult, subnetStatus] = await Promise.all([makeQuery(), getSubnetStatus()]);
+      const [queryResult, subnetStatus] = await Promise.all([
+        makeQuery(),
+        getSubnetStatus().catch(err => {
+          this.log.warn('Failed to fetch subnet keys. Error:', err);
+          return undefined;
```

Shouldn't we rethrow the error here?

---

The error is thrown in the `#verifyQuerySignature` function: https://github.com/dfinity/agent-js/blob/de5e3afa9c851c1f621a23f85d61918b83988fd3/packages/agent/src/agent/http/index.ts#L1021. Does it make sense?

Throwing an error in the `Promise.all` causes the other in-flight promises to be cancelled, which may not be what we want here